packages/flame_tiled/lib/src/renderable_layers/renderable_layer.dart

```diff
@@ -13,12 +18,66 @@ abstract class RenderableLayer<T extends Layer> {
   /// The parent [Group] layer (if it exists)
   final GroupLayer? parent;
 
-  RenderableLayer(
-    this.layer,
-    this.parent,
-    this.map,
-    this.destTileSize,
-  );
+  RenderableLayer({
+    required this.layer,
+    required this.parent,
+    required this.map,
+    required this.destTileSize,
+  });
+
+  /// [load] is a factory method to create [RenderableLayer] by type of [layer].
+  static Future<RenderableLayer> load({
+    required Layer layer,
+    required GroupLayer? parent,
+    required TiledMap map,
+    required Vector2 destTileSize,
+    required Camera? camera,
+    required Map<Tile, TileFrames> animationFrames,
+    required TiledAtlas atlas,
+    bool? ignoreFlip,
+  }) async {
+    if (layer is TileLayer) {
+      return FlameTileLayer.load(
+        layer: layer,
+        parent: parent,
+        map: map,
+        destTileSize: destTileSize,
+        animationFrames: animationFrames,
+        atlas: atlas.clone(),
+        ignoreFlip: ignoreFlip,
+      );
+    } else if (layer is ImageLayer) {
+      return FlameImageLayer.load(
+        layer: layer,
+        parent: parent,
+        camera: camera,
+        map: map,
+        destTileSize: destTileSize,
+      );
+    } else if (layer is ObjectGroup) {
+      return ObjectLayer.load(
+        layer,
+        map,
+        destTileSize,
+      );
+    } else if (layer is Group) {
+      final groupLayer = layer;
+      return GroupLayer(
+        layer: groupLayer,
+        parent: parent,
+        map: map,
+        destTileSize: destTileSize,
+      );
+    } else {
+      assert(false, '$layer layer is unsupported.');
```

Comment:
i don't think you want the assert, given that we have "unsupported layer" as the return type.

Reply:
I tried to decide whether I should touch this assertion or not.
I left this because the users can be conscious to some layers are not supported.

The assertion may be too strict but giving no warning is probably annoying to users like ...
 Why is it not working? I should report it to the issue tab.

As mentioned in [this comment](https://github.com/flame-engine/flame/pull/2310#issuecomment-1413876378),
> Make some logging systems that warn flame users about what went wrong.

It would be better if there is some way to warn users that doesn't break the code but can warn it.

What do you think?

Reply:
Agree that the assert probably doesn't give anything here, especially since this is an internal class.